src/main/scala/loamstream/util/BashScript.scala

```diff
@@ -26,10 +28,25 @@ object BashScript extends Loggable {
   /** Characters that should be escaped by prefixing with backslash */
   private val charsToBeEscaped: Set[Char] = Set('\\', '\'', '\"', '\n', '\r', '\t', '\b', '\f', ' ')
 
-  /** Escapes string for Bash. */
-  def escapeString(string: String): String = string.flatMap {
-    case c if charsToBeEscaped(c) => Seq('\\', c)
-    case c => Seq(c)
+  /** 
+   * Escapes string for Bash.
+   * Note new StringBuilder-oriented approach, which is significantly faster than the previous one that relied on
+   * String.flatMap.  Somewhat surprisingly, this method was a significant component of the graph-validation code's
+   * running time before switching to the StringBuilder approach.   
+   */
+  def escapeString(string: String): String = {
+    val builder = new StringBuilder(string.length * 2)
```

Comment:
Given that this was the case (that escaping strings was done often enough to show up), it might be worth considering making the `StringBuilder` a member of `BashScript` or - better still - a member of a companion object (if it doesn't need to be thread safe) and having it always resident for use, clearing and reusing, might save a lot of time constructing, destroying, GC'ing, etc.

Reply:
That's a good idea, if memory-management overhead related to the builder was significant.  I have no idea one way or another - this profiling exercise taught me to assume nothing! :)